src/Eryph.ConfigModel.Core.Validation/Validations.cs

```diff
@@ -54,30 +56,67 @@ public static class Validations
+                           Error.New($"The {valueName} cannot contain consecutive "
+                              + JoinItems("or", Seq(
+                                  Some("dots").Filter(_ => allowDots),
+                                  Some("hyphens").Filter(_ => allowHyphens),
+                                  Some("spaces").Filter(_ => allowSpaces)))
+                              + "."))
+                       .ToValidation()
+                   | guardnot(notEmpty(value) && value[0] is '.' or '-' or ' ',
+                           Error.New($"The {valueName} cannot start with a "
+                              + JoinItems("or", Seq(
+                                  Some("dot").Filter(_ => allowDots),
+                                  Some("hyphen").Filter(_ => allowHyphens),
+                                  Some("space").Filter(_ => allowSpaces)))
+                              + "."))
+                       .ToValidation()
+                   | guardnot(notEmpty(value) && value.Last() is '.' or '-' or ' ',
+                           Error.New($"The {valueName} cannot end with a "
+                              + JoinItems("or", Seq(
+                                  Some("dot").Filter(_ => allowDots),
+                                  Some("hyphen").Filter(_ => allowHyphens),
+                                  Some("space").Filter(_ => allowSpaces)))
+                              + "."))
+                       .ToValidation()
         select value;
 
-    public static Validation<Error, string> ValidatePath(
+    public static Validation<Error, string> ValidateWindowsPath(
         string? value,
         string valueName) =>
+        // This code intentionally does not use any System.IO methods
+        // as their behavior is OS dependent.
         from nonEmptyValue in ValidateNotEmpty(value, valueName)
-        from _ in guardnot(Path.GetInvalidPathChars().Intersect(nonEmptyValue).Any(),
-                          Error.New($"The {valueName} must be a valid path but contains invalid characters."))
+        from _ in guardnot(WindowsPath.GetInvalidPathChars().Intersect(nonEmptyValue).Any(),
+                          Error.New($"The {valueName} must be a valid Windows path but contains invalid characters."))
+                      .ToValidation()
+                  | guardnot(nonEmptyValue.Contains('/'),
+                          Error.New($"The {valueName} must only contain Windows directory separators."))
                       .ToValidation()
                   | guardnot(nonEmptyValue.Length > 260,
-                          Error.New($"The {valueName} must be a valid path but contains more than 260 characters."))
+                          Error.New($"The {valueName} must be a valid Windows path but contains more than 260 characters."))
+                      .ToValidation()
+        from __ in guard(nonEmptyValue.StartsWith(@"\\") || DriveRootRegex.IsMatch(nonEmptyValue),
```

Comment:
actually a network path may be allowed

Reply:
Something like `\\TESTSRV\testshare` is allowed due to the `nonEmptyValue.StartsWith(@"\\")` (and covered by a unit test). Is another kind of network path we need to allow?